buildpacks/java/java/build.go

```diff
@@ -53,7 +53,7 @@ func (b Build) Build(context libcnb.BuildContext) (libcnb.BuildResult, error) {
 
 	functionLayer := NewFunction(context.Application.Path,
 		WithLogger(b.Logger),
-		WithFunctionClass(functionClass, isFuncDefDefault),
+		WithFunctionClass(functionClass, isFuncDefDefault, e.Metadata["func_yaml_function_name"].(string)),
```

Comment:
I think what you want here is to make `func_yaml_function_name` to be the *default function* if it's not empty and `isDefaultFuncDefault` is false.

Reply:
I also noticed I mis-named those variables `isFuncDefDefault` and `isDefaultFuncDefault`.

They should have been `isFuncDefDefined` and `isDefaultFuncDefined`.
`cr.resolve` returns whether it was explicitly set or not. (`false` == user did not set the value)